hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/test/java/org/apache/hadoop/yarn/server/federation/store/impl/TestZookeeperFederationStateStore.java

```diff
@@ -171,38 +203,293 @@ public void testMetricsInited() throws Exception {
+    FederationStateStore stateStore = this.getStateStore();
+    int seqNum = stateStore.incrementDelegationTokenSeqNum();
+    identifier.setSequenceNumber(seqNum);
+    Long renewDate = Time.now();
+
+    // Store new rm-token()
+    RouterStoreToken storeToken = RouterStoreToken.newInstance(identifier, renewDate);
+    RouterRMTokenRequest request = RouterRMTokenRequest.newInstance(storeToken);
+    RouterRMTokenResponse routerRMTokenResponse = stateStore.storeNewToken(request);
+    Assert.assertNotNull(routerRMTokenResponse);
+
+    // We are ready to update some data renewDate2 & sequenceNumber2
+    Long renewDate2 = Time.now();
+    int sequenceNumber2 = stateStore.incrementDelegationTokenSeqNum();
+    identifier.setSequenceNumber(sequenceNumber2);
+
+    // Update rm-token
+    RouterStoreToken paramStoreToken = RouterStoreToken.newInstance(identifier, renewDate2);
+    RouterRMTokenRequest updateTokenRequest = RouterRMTokenRequest.newInstance(paramStoreToken);
+    RouterRMTokenResponse updateTokenResponse = stateStore.updateStoredToken(updateTokenRequest);
+    Assert.assertNotNull(updateTokenResponse);
+    RouterStoreToken updateTokenResp = updateTokenResponse.getRouterStoreToken();
+
+    // Get delegationToken Path
+    String nodeName = ROUTER_RM_DELEGATION_TOKEN_PREFIX + identifier.getSequenceNumber();
+    String nodePath = getNodePath(ZNODE_DT_PREFIX, nodeName);
+
+    // Check if the path exists, we expect the result to exist.
+    assertTrue(isExists(nodePath));
+
+    // Check whether the token (paramStoreToken)
+    // We generated is consistent with the data stored in zk.
+    // We expect data to be consistent.
+    RouterStoreToken zkRouterStoreToken = getStoreTokenFromZK(nodePath);
+    assertNotNull(zkRouterStoreToken);
+    assertEquals(paramStoreToken, zkRouterStoreToken);
+    assertEquals(updateTokenResp, zkRouterStoreToken);
   }
 
-  @Test(expected = NotImplementedException.class)
-  public void testRemoveStoredToken() throws IOException, YarnException {
-    super.testRemoveStoredToken();
+  @Test
+  public void testRemoveStoredToken() throws YarnException, IOException {
+
+    // We manually generate the DelegationToken,
+    // and then call the StoreNewToken method to store the Token in zk.
+    RMDelegationTokenIdentifier identifier = new RMDelegationTokenIdentifier(
+        new Text("owner2"), new Text("renewer2"), new Text("realuser2"));
+    FederationStateStore stateStore = this.getStateStore();
+    int seqNum = stateStore.incrementDelegationTokenSeqNum();
+    identifier.setSequenceNumber(seqNum);
+    Long renewDate = Time.now();
+
+    // Store new rm-token
+    RouterStoreToken storeToken = RouterStoreToken.newInstance(identifier, renewDate);
+    RouterRMTokenRequest request = RouterRMTokenRequest.newInstance(storeToken);
+    RouterRMTokenResponse routerRMTokenResponse = stateStore.storeNewToken(request);
+    Assert.assertNotNull(routerRMTokenResponse);
```

Comment:
As you statically imported assertTrue and others, import this too.

Reply:
I will fix it.